ast/desugar/Desugar.cc

```diff
@@ -797,7 +796,7 @@ ExpressionPtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what)
                     sendargs.emplace_back(std::move(kwargs));
                     ExpressionPtr res;
                     if (block == nullptr && !anonymousBlockPass) {
-                        res = MK::Send(loc, MK::Magic(loc), core::Names::callWithSplat(), locZeroLen, 4,
+                        res = MK::Send(loc, MK::Magic(loc), core::Names::callWithSplat(), send->methodLoc, 4,
```

Comment:
Is this change really innocuous?  I guess the rationale is that the `Send` we're creating here ought to be analogous to the `Send` that textually appears, even if the new `Send` is using some Sorbet-internal thing to do the work.

Reply:
This is what enables the correct `^^^` lines below.

Before, it would see that the funLoc was zero and decided to report the "method does not exist" error on the `callLoc`, which is what the first argument to the `MK::Send` is, which uses `loc`, indicating that it's the complete parser::Send node's range.

So yes this change is pretty innocuous in my mind, and in fact more correct than it was previously.